services/graph/pkg/identity/ldap.go

```diff
@@ -802,3 +882,129 @@ func replaceDN(fullDN *ldap.DN, newDN string) (string, error) {
 
 	return newDN, nil
 }
+
+// CreateLDAPGroupByDN is a helper method specifically intended for creating a "system" group
+// for managing locally disabled users on service startup
+func (i *LDAP) CreateLDAPGroupByDN(dn string) error {
+	ar := ldap.NewAddRequest(dn, nil)
+
+	attrs := map[string][]string{
+		"objectClass": {"groupOfNames", "top"},
+		"member":      {""},
+	}
+
+	for attrType, values := range attrs {
+		ar.Attribute(attrType, values)
+	}
+
+	return i.conn.Add(ar)
+}
+
+func (i *LDAP) disableUser(logger log.Logger, userDN string) (err error) {
+	group, err := i.getEntryByDN(i.localUserDisableGroupDN, []string{_ldapGroupMemberAttribute}, _ldapGroupOfNamesAttribute)
+
+	if err != nil {
+		return err
+	}
+
+	mr := ldap.ModifyRequest{DN: group.DN}
+	mr.Add(_ldapGroupMemberAttribute, []string{userDN})
+
+	err = i.conn.Modify(&mr)
+	var lerr *ldap.Error
+	if errors.As(err, &lerr) {
+		// If the user is already in the group, just log a message and return
+		if lerr.ResultCode == ldap.LDAPResultAttributeOrValueExists {
+			logger.Info().Msg("User already in group for disabled users")
+			return nil
+		}
+	}
+
+	return err
+}
+
+func (i *LDAP) enableUser(logger log.Logger, userDN string) (err error) {
+	group, err := i.getEntryByDN(i.localUserDisableGroupDN, []string{_ldapGroupMemberAttribute}, _ldapGroupOfNamesAttribute)
+
+	if err != nil {
+		return err
+	}
+
+	mr := ldap.ModifyRequest{DN: group.DN}
+	mr.Delete(_ldapGroupMemberAttribute, []string{userDN})
+
+	err = i.conn.Modify(&mr)
+	var lerr *ldap.Error
+	if errors.As(err, &lerr) {
+		// If the user is not in the group, just log a message and return
+		if lerr.ResultCode == ldap.LDAPResultNoSuchAttribute {
+			logger.Info().Msg("User was not in group for disabled users")
+			return nil
+		}
+	}
+
+	return err
+}
+
+func (i *LDAP) userEnabledByAttribute(user *ldap.Entry) bool {
+	enabledAttribute := booleanOrNil(user.GetEqualFoldAttributeValue(i.userAttributeMap.accountEnabled))
+
+	if enabledAttribute == nil {
+		return true
+	}
+
+	return *enabledAttribute
+}
+
+func (i *LDAP) usersDisabledByGroup(users []string) (usersEnabledState map[string]bool, err error) {
```

Comment:
I think the naming of this method is a bit misleading. It returns a map[string]bool, where the the value is true if the user is enabled (or more precisely: "not disabled"). The name of the method seems to imply the opposite. Or is it just me?